codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-endpoint-discovery-async.java

```diff
@@ -193,12 +193,12 @@ public CompletableFuture<TestDiscoveryIdentifiersRequiredResponse> testDiscovery
                     .overrideConfiguration().flatMap(AwsRequestOverrideConfiguration::credentialsIdentityProvider)
                     .orElseGet(() -> clientConfiguration.option(AwsClientOption.CREDENTIALS_IDENTITY_PROVIDER))
                     .resolveIdentity();
-                endpointFuture = identityFuture.thenApply(credentials -> {
+                endpointFuture = identityFuture.thenCompose(credentials -> {
```

Comment:
Should we propagate exception cancellation? What happens if endpoint discovery takes longer than the configured client execution timeout? Would the request fail properly?

Reply:
hmm.. good catch. I will propogate cancellation exceptions and terminate this in the event of a client exception / interrupt

Reply:
I will add an interrupted exception

Reply:
I meant if users have client execution timeout enabled and if endpoint discovery takes longer than the configured timeout, the returned future will be cancelled, but will endpoint discovery future also be cancelled?

Reply:
I think we just need ` CompletableFutureUtils.forwardExceptionTo(executeFuture, endpointFuture );` on line 277. 

Can we write a test to verify that?

Example: https://github.com/aws/aws-sdk-java-v2/blob/master/services/s3/src/test/java/software/amazon/awssdk/services/s3/internal/crt/CopyObjectHelperTest.java#L314

Reply:
Discussed offline. Doing this as a follow up item